src/legacy/components/OrgUnitsSelectorFilter.js

```diff
@@ -41,44 +38,33 @@ class OrgUnitsFilter extends React.Component {
         },
     };
 
-    componentDidMount = () => {
-        return getOrgUnitsRoots().then(roots => this.setState({ roots }));
-    };
-
     componentWillReceiveProps(newProps) {
         if (newProps.selected !== this.props.selected) this.fieldValue = this.getCompactFieldValue(newProps.selected);
     }
 
-    openDialog = () => {
+    openDialog() {
         this.setState({ dialogOpen: true, selected: this.props.selected });
-    };
+    }
 
-    closeDialog = () => {
+    closeDialog() {
         this.setState({ dialogOpen: false });
-    };
+    }
 
     onChange(selected) {
         this.setState({ selected });
     }
 
-    applyAndClose = () => {
-        this.props.onChange(this.state.selected);
-        this.closeDialog();
-    };
+    async applyAndClose() {
+        const { d2 } = this.context;
+        const orgUnitIds = getOrgUnitsPaths(this.state.selected).map(path => _.last(path.split("/")));
```

Comment:
The path to id conversion is heavily used, maybe a helper method instead of repeating the last/split combination could be helpful.

```
export function cleanOrgUnitPath(orgUnitPath?: string): string {
    return _(orgUnitPath).split("/").last() ?? "";
}

export function cleanOrgUnitPaths(orgUnitPaths: string[]): string[] {
    return orgUnitPaths.map(cleanOrgUnitPath);
}
```

Reply:
I'm not sure what you are asking, should I add the above code in helpers and then use it in line 59 and in the other places that use the same code? And which function?